scanpy/preprocessing/_highly_variable_genes.py

```diff
@@ -486,14 +487,16 @@ def highly_variable_genes(
     else:
         sanitize_anndata(adata)
         batches = adata.obs[batch_key].cat.categories
-        df = []
+        dfs = []
         gene_list = adata.var_names
         for batch in batches:
             adata_subset = adata[adata.obs[batch_key] == batch]
 
             # Filter to genes that are in the dataset
             with settings.verbosity.override(Verbosity.error):
-                filt = filter_genes(adata_subset, min_cells=1, inplace=False)[0]
+                filt, _ = filter_genes(
+                    _get_obs_rep(adata_subset, layer=layer), min_cells=1, inplace=False
+                )
```

Comment:
this has been broken the whole time?

Reply:
I bet we didn't notice since the nonzero values are likely pretty similar, if not the same, for common use cases